### ISSUE TITLE
Updating custom tags annotation for services does not delete old tags

### ISSUE THREAD

**Issue by faraazkhan** [kubernetes/kubernetes#60828]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
 /kind bug
> /kind feature

**What happened**:
When a tag is applied and then removed using the `service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags` annotation, the tag  continues to exist on the AWS ELB. 

**What you expected to happen**:
Additional tags applied to the ELB should always match the annotation. 

**How to reproduce it (as minimally and precisely as possible)**:
Create a simple deployment and service using the manifest below and `kubectl apply -f`
```
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
spec:
  selector:
    matchLabels:
      app: nginx
  replicas: 1
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: name
        image: nginx
        imagePullPolicy: Always
        resources:
          requests:
            cpu: 100m
            memory: 200Mi
        ports:
        - containerPort: 80

---
apiVersion: v1
kind: Service
metadata:
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags: "app=nginx,costcenter=123,env=nonprod"
  name:  nginx-lb
spec:
  ports:
  - port: 80
    targetPort: 80
  selector:
    app: nginx
  type: LoadBalancer
```
Then modify the manifest, as shown below, by removing the env tag from the annotation, then `kubectl apply -f` again
```
---
apiVersion: v1
kind: Service
metadata:
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags: "app=nginx,costcenter=123"
  name:  nginx-lb
spec:
  ports:
  - port: 80
    targetPort: 80
  selector:
    app: nginx
  type: LoadBalancer
```
Navigate to the AWS console and notice how the `env` tag continues to exist on the ELB, even though it was removed from the annotation. 

**Anything else we need to know?**:

**Environment**: 
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-09T21:51:06Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T11:55:20Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}

```
- Cloud provider or hardware configuration: 
```

[global]
DisableSecurityGroupIngress=True
ElbSecurityGroup=sg-xxxxx
```

- OS (e.g. from /etc/os-release): 
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

- Kernel (e.g. `uname -a`):
```
Linux abpauva058154 3.10.0-693.17.1.el7.x86_64 #1 SMP Thu Jan 25 20:13:58 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools: ` Custom Terraform`
- Others:

**Comment by WanLinghao** [comment]
can you please add more information about how you remove env tag?

**Comment by faraazkhan** [comment]
I used `kubectl apply -f` on the manifests provided above. Notice the list of tags on the second manifest doesn’t have the env tag. @wangLinghao

**Comment by WanLinghao** [comment]
It works well on my k8s cluster.Here's my version info

```
Client Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0-alpha.1.3426+01504f66e3122a-dirty", GitCommit:"01504f66e3122ae3eca33c24a6450a9112eb8d87", GitTreeState:"dirty", BuildDate:"2018-03-06T05:41:01Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0-alpha.1.3426+01504f66e3122a-dirty", GitCommit:"01504f66e3122ae3eca33c24a6450a9112eb8d87", GitTreeState:"dirty", BuildDate:"2018-03-06T05:41:01Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by faraazkhan** [comment]
Hey @WanLinghao so are you saying this is resolved in 1.10. Because I can reproduce it pretty easily on 1.9.3. 
So after first creating the service with the `env` tag using the manifest above and `kubectl apply`, I then modify the manifest to remove the `env` tag from the annotation. 
The result of `kubectl describe svc nginx-lb` is pasted below:
```
apiVersion: v1
kind: Service
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"Service","metadata":{"annotations":{"service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags":"app=nginx,costcenter=123"},"name":"nginx-lb","namespace":"default"},"spec":{"ports":[{"port":80,"targetPort":80}],"selector":{"app":"nginx"},"type":"LoadBalancer"}}
    service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags: app=nginx,costcenter=123
  creationTimestamp: 2018-03-22T19:25:28Z
  name: nginx-lb
  namespace: default
  resourceVersion: "1326870"
  selfLink: /api/v1/namespaces/default/services/nginx-lb
  uid: c7998724-2e06-11e8-8e40-0eba86e0d536
spec:
  clusterIP: 10.3.0.65
  externalTrafficPolicy: Cluster
  ports:
  - nodePort: 31725
    port: 80
    protocol: TCP
    targetPort: 80
  selector:
    app: nginx
  sessionAffinity: None
  type: LoadBalancer
status:
  loadBalancer:
    ingress:
    - hostname: ac79987242e0611e88e400eba86e0d53-222479116.us-east-1.elb.amazonaws.com
```
Notice how the annotation no longer has the env tag. But when I look at the ELB, `aws elb describe-tags --load-balancer-name ac79987242e0611e88e400eba86e0d53` . I can still see the tag there, see output below:
```
{
    "TagDescriptions": [
        {
            "LoadBalancerName": "ac79987242e0611e88e400eba86e0d53",
            "Tags": [
                {
                    "Key": "app",
                    "Value": "nginx"
                },
                {
                    "Key": "kubernetes.io/service-name",
                    "Value": "default/nginx-lb"
                },
                {
                    "Key": "kubernetes.io/cluster/faraaz-nonprod",
                    "Value": "owned"
                },
                {
                    "Key": "env",
                    "Value": "nonprod"
                },
                {
                    "Key": "costcenter",
                    "Value": "123"
                }
            ]
        }
    ]
}
```
Output of `kubectl version` below:
```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.5", GitCommit:"f01a2bf98249a4db383560443a59bed0c13575df", GitTreeState:"clean", BuildDate:"2018-03-19T19:38:53Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T11:55:20Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by faraazkhan** [comment]
Also just looking through the code, it seems that while we sync various things like subnets, listeners etc. on the ELBs, when it comes to tags, we are always simply adding additional tags: https://github.com/kubernetes/kubernetes/blob/eff07b6f555b6b5ef124a50c37be7a632e11c2ba/pkg/cloudprovider/providers/aws/aws_loadbalancer.go#L1149. Add to that the fact that the ELB interface does not even reference `RemoveTags` here: https://github.com/kubernetes/kubernetes/blob/eff07b6f555b6b5ef124a50c37be7a632e11c2ba/pkg/cloudprovider/providers/aws/aws.go#L314 I am starting to think if this feature (removal of tags when annotation is updated) even exists.
